File: Raven.Client.Java/src/main/java/net/ravendb/client/connection/IDocumentStoreReplicationInformer.java
package net.ravendb.client.connection;

import net.ravendb.abstractions.replication.ReplicationDestination;


public interface IDocumentStoreReplicationInformer extends IReplicationInformerBase<ServerClient> {

  public ReplicationDestination[] getFailoverServers();

  public void setFailoverServers(ReplicationDestination[] failoverServers);

  /**
   * Updates replication information if needed
   */
  @Override
  public void updateReplicationInformationIfNeeded(ServerClient serverClient);

  /**
   * Updates replication information
   */
  @Override
  public void refreshReplicationInformation(ServerClient serverClient);

}

File: src/Raven.NewClient/Connection/IDocumentStoreReplicationInformer.cs
using System.Threading.Tasks;

using Raven.Abstractions.Replication;
using Raven.NewClient.Client.Connection.Async;

namespace Raven.NewClient.Client.Connection
{
    public interface IDocumentStoreReplicationInformer : IReplicationInformerBase<ServerClient>
    {
        /// <summary>
        /// Failover servers set manually in config file or when document store was initialized
        /// </summary>
        ReplicationDestination[] FailoverServers { get; set; }

        /// <summary>
        /// Updates replication information if needed
        /// </summary>
        Task UpdateReplicationInformationIfNeededAsync(AsyncServerClient serverClient);
    }
}
